### PR TITLE
feat: Add Citations to Grounding with Gemini Notebook

### LINKED ISSUE #641: [Bug]:  Add `grounding_metadata` to Grounding with Gemini Notebook
### File Name

gemini/grounding/intro-grounding-gemini.ipynb

### What happened?

Add print out of `canidate.grounding_metadata` once https://github.com/googleapis/python-aiplatform/pull/3712 is released to the SDK

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by holtskinner**
# Description

- Added citation output for Gemini Grounding.
- Updated model to Gemini 1.5 Pro
- Added more explanations about Grounding with Google Search Allowlist
- Fixed some copy/paste mistakes from PaLM Grounding notebook

Fixes #641 🦕

### CODE REVIEW

File: gemini/grounding/intro-grounding-gemini.ipynb [hunk 1]
```diff
@@ -438,14 +487,14 @@
...
-   "execution_count": 7,
+   "execution_count": null,
    "metadata": {},
    "outputs": [],
    "source": [
     "DATA_STORE_PROJECT_ID = PROJECT_ID  # @param {type:\"string\"}\n",
     "DATA_STORE_REGION = \"global\"  # @param {type:\"string\"}\n",
     "# Replace this with your data store ID from Vertex AI Search\n",
-    "DATA_STORE_ID = \"your-data-store-id_1234567890123\"  # @param {type:\"string\"}"
+    "DATA_STORE_ID = \"google-cloud-docs_1695249676245\"  # @param {type:\"string\"}"
```
Comment (koverholt) [inline gemini/grounding/intro-grounding-gemini.ipynb:497]: Do you need to reset this to the dummy/placeholder value?

**Review by koverholt — APPROVED** [review]
LGTM!

File: gemini/grounding/intro-grounding-gemini.ipynb (hunk 1 above)
Comment (holtskinner) [reply gemini/grounding/intro-grounding-gemini.ipynb:497]: Probably a good idea.

Comment (holtskinner) [inline gemini/grounding/intro-grounding-gemini.ipynb:497]:
```suggestion
    "DATA_STORE_ID = \"your-data-store-id_1234567890123\"  # @param {type:\"string\"}"
```

**Pull request merged**